sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -105,10 +105,22 @@ case class InsertIntoHadoopFsRelationCommand(
     }
 
     val jobId = java.util.UUID.randomUUID().toString
+    // For dynamic partition overwrite, FileOutputCommitter's output path is staging path, files
+    // will be renamed from staging path to final output path during commit job
+    val committerOutputPath = if (dynamicPartitionOverwrite) {
+      FileCommitProtocol.getStagingDir(outputPath.toString, jobId)
+        .makeQualified(fs.getUri, fs.getWorkingDirectory)
+    } else if (staticPartitions.size == partitionColumns.length) {
+      val defaultLocation = qualifiedOutputPath.suffix(
+        "/" + PartitioningUtils.getPathFragment(staticPartitions, partitionColumns)).toString
+      customPartitionLocations.getOrElse(staticPartitions, defaultLocation)
```

Comment:
Here will change the value of committerOutputPath in static partition insert to partition location. It's reasonable.
This value is only used in 
 1.    FileOutputFormat.setOutputPath(job, new Path(outputSpec.outputPath))
 2.   Assigned to WriteJobDescription's path parameter, it's not used at all